### ISSUE TITLE
Constant 100% CPU utilisation

### ISSUE THREAD

**Issue by GaborPeto** [input-output-hk/jormungandr#988]
**Describe the bug**
After the node was started I monitored my system's CPU utilisation.
During the initial bootstrap phase CPU usage was fluctuating between 25 and 75%.
After the bootstrap was finished and an initial ` block successfully applied` message I saw ` connecting to peer` messages and a constant 100% CPU utilisation on all 4 of my vCPUs (for at least 30mins without the slightest drop). When the node was shut down the CPU usage was still on 100% and never dropped and the node did not receive a single block from the network.

*Note*
I also tested 0.6.3 on windows and at no point, during nor after the bootstrap the CPU utilisation was above 2%.

**Mandatory Information**
1. `jcli --full-version` output; jcli 0.6.3 (HEAD-cf872d4, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)]

2. `jormungandr --full-version` output; jormungandr 0.6.3 (HEAD-cf872d4, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)]

**To Reproduce**
Start the node using 
```
./jormungandr --config node-config.yaml --genesis-block-hash adbdd5ede31637f6c9bad5c271eec0bc3d0cb9efb86a5b913bb55cba549d0770 --log-level=info
```

**Expected behavior**
Slight CPU usage spikes are acceptable however the node should not run on 100% CPU utilisation for 30 mins.

**Additional context**
`node-config.yaml`:
```
log:
  format: "plain"
  level: "info"
  output: "stderr"
p2p:
  listen_address: "/ip4/0.0.0.0/tcp/9000"
  public_address: "/ip4/178.62.25.199/tcp/9000"
  topics_of_interest:
    blocks: "normal"
    messages: "low"
  trusted_peers:
    - address: "/ip4/35.199.15.206/tcp/9000"
      id: ed25519_pk14jkw8jnkaknjjde9zluzz0mt332ypxehv2yc99aa3zlumm0qwc7ssmvq06
    - address: "/ip4/3.115.194.22/tcp/3000"
      id: ed25519_pk1npsal4j9p9nlfs0fsmfjyga9uqk5gcslyuvxy6pexxr0j34j83rsf98wl2
    - address: "/ip4/13.113.10.64/tcp/3000"
      id: ed25519_pk16pw2st5wgx4558c6temj8tzv0pqc37qqjpy53fstdyzwxaypveys3qcpfl
    - address: "/ip4/52.57.214.174/tcp/3000"
      id: ed25519_pk1v4cj0edgmp8f2m5gex85jglrs2ruvu4z7xgy8fvhr0ma2lmyhtyszxtejz
    - address: "/ip4/3.120.96.93/tcp/3000"
      id: ed25519_pk10gmg0zkxpuzkghxc39n3a646pdru6xc24rch987cgw7zq5pmytmszjdmvh
    - address: "/ip4/52.28.134.8/tcp/3000"
      id: ed25519_pk1unu66eej6h6uxv4j4e9crfarnm6jknmtx9eknvq5vzsqpq6a9vxqr78xrw
    - address: "/ip4/13.52.208.132/tcp/3000"
      id: ed25519_pk15ppd5xlg6tylamskqkxh4rzum26w9acph8gzg86w4dd9a88qpjms26g5q9
    - address: "/ip4/54.153.19.202/tcp/3000"
      id: ed25519_pk1j9nj2u0amlg28k27pw24hre0vtyp3ge0xhq6h9mxwqeur48u463s0crpfk
rest:
  listen: "127.0.0.1:3101"
storage: "storage/"
```

**Setup**:
4vcpu / 8 GB Memory / 160 GB Disk / - Ubuntu 18.04.3 (LTS) x64

**Comment by mark-stopka** [comment]
Duplicate of #987

**Comment by mark-stopka** [comment]
@GaborPeto, is it syncing with the chain thru the high CPU utilization for you though?

**Comment by GaborPeto** [comment]
@mark-stopka reported around the same time, different descriptions.
My node was not syncing.
